src/extensions/agent_modules/device_update_workflow/module/src/device_update_workflow_module.c

```diff
@@ -104,35 +104,38 @@ int DeviceUpdateWorkflowModule_DoWork(ADUC_AGENT_MODULE_HANDLE handle)
  */
 ADUC_AGENT_MODULE_HANDLE DeviceUpdateWorkflowModule_Create()
 {
-    bool ret = false;
     ADUC_WorkflowData *workflowData =  NULL;
+
     ADUC_AGENT_MODULE_HANDLE handle = NULL;
-    ADUC_AGENT_MODULE_INTERFACE* interface = NULL;
+    ADUC_AGENT_MODULE_INTERFACE* tmp = NULL;
 
-    ADUC_ALLOC(interface);
+    tmp = calloc(1, sizeof(*tmp));
+    if (tmp == NULL)
+    {
+        goto done;
+    }
 
     // TODO (nox-msft) - pass in launchArgs
-    ret = AzureDeviceUpdateCoreInterface_Create((void**)&workflowData, 0, NULL);
-    if (!ret)
+    if (!AzureDeviceUpdateCoreInterface_Create((void**)&workflowData, 0, NULL))
     {
         goto done;
     }
 
     workflowData->CommunicationChannel = ADUC_CommunicationChannelType_MQTTBroker;
-    interface->moduleData = workflowData;
-    interface->destroy = DeviceUpdateWorkflowModule_Destroy;
-    interface->getContractInfo = DeviceUpdateWorkflowModule_GetContractInfo;
-    interface->doWork = DeviceUpdateWorkflowModule_DoWork;
-    interface->initializeModule = DeviceUpdateWorkflowModule_Initialize;
-    interface->deinitializeModule = DeviceUpdateWorkflowModule_Deinitialize;
 
-    handle = (ADUC_AGENT_MODULE_HANDLE)interface;
+    tmp->moduleData = workflowData;
```

Comment:
Set workflowData to NULL here and free(workflowData) after done: ? (follow your pattern)

Reply:
good catch. set to NULL and doing ADUC_WorkflowData_Uninit(workflowData) and free(workflowData) unconditionally after done: label